sktime/param_est/compose/_func_fitter.py

```diff
@@ -0,0 +1,131 @@
+    param : str
+        The name of the parameter to set.
+    func : callable (X: X_type, **kwargs) -> Any
+        The callable to use for the parameter estimation. This will be
+        passed the same arguments as estimator, with args and kwargs
+        forwarded.
+    kw_args : dict, default=None
+        Dictionary of additional keyword arguments to pass to func.
+    X_type : str, one of "pd.DataFrame, pd.Series, np.ndarray", or list thereof
+        default = ["pd.DataFrame", "pd.Series", "np.ndarray"]
+        list of types that func is assumed to allow for X (see signature above)
+        if X passed to transform/inverse_transform is not on the list,
+            it will be converted to the first list element before passed to funcs
+
+    See Also
+    --------
+    sktime.param_est.plugin.PluginParamsForecaster :
+        Plugs parameters from a parameter estimator into a forecaster.
+    sktime.forecasting.compose.MultiplexForecaster :
+        MultiplexForecaster for selecting among different models.
+
+    Examples
+    --------
+    This class could be used to contruct a parameter estimator that
+    selects a forecaster based on the input data's length. The
+    selected forecaster can be stored in the `selected_forecaster_`
+    attribute, which can be then passed down to a
+    ``MultiplexForecaster`` via a ``PluginParamsForecaster``.
+
+    >>> import numpy as np
+    >>> param_est = FunctionParamFitter(
+    ...     param="selected_forecaster",
+    ...     func=(
+    ...         lambda X, threshold: "naive-seasonal"
+    ...         if len(X) >= threshold
+    ...         else "naive-last"
+    ...     ),
+    ...     kw_args={"threshold": 7},
+    ... )
+    >>> param_est.fit(np.asarray([1, 2, 3, 4]))
+    >>> param_est.selected_forecaster_
+    'naive-last'
+    >>> param_est.fit(np.asarray([1, 2, 3, 4, 5, 6, 7]))
+    >>> param_est.selected_forecaster_
+    'naive-seasonal'
+    """
+
+    _tags = {
+        "X_inner_mtype": ALL_TIME_SERIES_MTYPES,
+        "scitype:X": ["Series", "Panel", "Hierarchical"],
+        "capability:missing_values": True,
+        "capability:multivariate": False,
+    }
+
+    def __init__(self, param, func, kw_args=None, X_type=None):
+        self.param = param
+        self.func = func
+        self.kw_args = kw_args
+        self.X_type = X_type
+        super().__init__()
+
+        if X_type is not None:
+            self.set_tags(X_inner_mtype=X_type)
+
+    def _fit(self, X):
+        """Fit estimator and estimate parameters.
+
+        private _fit containing the core logic, called from fit
+
+        Writes to self:
+            Sets fitted model attributes ending in "_".
+
+        Parameters
+        ----------
+        X : guaranteed to be of a type in self.get_tag("X_inner_mtype")
+            Time series to which to fit the estimator.
+
+        Returns
+        -------
+        self : reference to self
+        """
+        param = self.param.rstrip("_") + "_"
```

Comment:
@fkiraly Just double checking in case you missed this one. This decision is not explicitly documented or explained anywhere to the user. Still, this line implicitly makes `'some_param'`, `'some_param_'`, and even `'some_param___'` to be treated as the same parameter (i.e. they are all automatically converted to `'some_param_'` here). There are a few options to consider here:
1. Allow the user to pass any variation of `'some_param'`, `'some_param_'`, `'some_param___'`, etc. and automatically strip and convert to `'some_param_'` (**the current implementation in the PR**)
2. Only allow parameter name without a trailing `_`. i.e., raise a value error if `param.endswith('_')`
3. Require the user to pass the parameter name with the trailing `_`
4. Something else...?

I think that the first two options make the most sense but I'm more inclined towards the first one as it creates the least friction for end users (i.e., _it just works_)

Reply:
hm, I think it should be consistent with the name in `get_fitted_params`, so the current implementation makes sense.